src/taco-utils/promisesUtils.ts

```diff
@@ -0,0 +1,31 @@
+﻿/**
+﻿ *******************************************************
+﻿ *                                                     *
+﻿ *   Copyright (C) Microsoft. All rights reserved.     *
+﻿ *                                                     *
+﻿ *******************************************************
+﻿ */
+
+/// <reference path="../typings/lodash.d.ts" />
+
+"use strict";
+
+import _ = require("lodash");
+
+module TacoUtility {
+    export class PromisesUtils {
+        public static executeAfter<T>(valueOrPromise: T, afterCallback: { (): void }, failCallback?: { (reason: any): void }): T {
```

Comment:
Yes, for the synchronously case alone this method is useless.

The reason I use this method is because I wanted to have a single method that works both for sync and async cases instead of using telemetry.timeSync() and telemetry.timeAsync(), and this is the only way I could think of how to do it. If you come up with a different way of implementing this, I'm totally fine with changing it.

The other option would be to use 2 different methods with the implementations that you suggested:
telemetry.timeSync(): var result = someSyncCalculation(); afterCallback();
telemetry.timeAsync(): valueOrPromise.finally(afterCallback).fail(failCallback)

But I personally like having a single method for both cases. What do you think we should do?